ACD-618 (#30)
Actindo-AG/Documentation: Channels/Integration/02_ManageVariantSets.md

[!!User interface Variant sets](../UserInterface/to_be_completed)
[!!Manage the variant sets in PIM](../../PIM/Integration/07_ManageVariantSets.md)

# Manage a variant set

> The variant sets in *Omni-Channel* are based on the same data model as in the Actindo *PIM* module. However, since the *Omni-Channel* and the Actindo *PIM* module are independent modules and *Omni-Channel* can also be used without the Actindo *PIM* module, the variant sets for products can also be managed within the *Omni-Channel* module.    

Reviewer: ```suggestion
The variant sets in *Omni-Channel* are based on the same data model as in the Actindo *PIM* module. However, since the *Omni-Channel* and the Actindo *PIM* modules are independent of each other and can be used separately, the variant sets for products can also be managed within the *Omni-Channel* module.    
```

If both the *Omni-Channel* and the Actindo *PIM* modules are used, there is a predefined mapping that automatically applies the variant sets from *PIM* to *Omni-Channel*. In this case, you do not have to manage the variant sets in *Omni-Channel*, but in the Actindo *PIM* module, see [Manage the variant sets in PIM](../../PIM/Integration/07_ManageVariantSets.md).

## Create a variant set

> Create a variant set to be able to create variants to products. A variant set is always assigned to a certain attribute set. When a variant set is created to an attribute set, you can create variants to all products with this attribute set. You must assign at least one defining attribute to a variant set. Moreover, you can assign further defining and changeable attributes.   

Reviewer: ```suggestion
Create a variant set to be able to create variants to products. A variant set is always assigned to a certain attribute set. When a variant set is assigned to an attribute set, you can create variants to all products with this attribute set. You must assign at least one defining attribute to a variant set. Moreover, you can assign further defining and changeable attributes.   
```

  The *Formulas* section with the *Formula for SKU* field is displayed.

  > [Info] You need to specify defining and changeable attributes for the variant set at a later step. Be aware that only those attributes assigned to the selected attribute set will be available.

  ![Formulas](../../Assets/Screenshots/Channels/Settings/VariantSets/Formulas.png "[Formulas]")

4. Enter a formula for the variants SKU in the *Formula for SKU* field using the displayed placeholders. By default, you can enter the formula **{master}-{L}**.   

  > [Info] The value of the defining attributes can be included into the SKU formula. When you have added a defining attribute, the corresponding placeholder for this attribute is displayed in the *Formulas* section.

5. For the next steps to create a variant set, follow the procedures below:   
  - [Add defining attributes](#add-defining-attributes)
  - [Add changeable attributes](#add-changeable-attributes)

  > [Info] At least one defining attribute is required to create a variant set. Changeable attributes can be selected optionally.

### Add defining attributes

> You have to select at least one defining attribute to create a variant set. You can only select attributes as defining attributes, that are neither multi-language nor multi-channel and assigned to the selected attribute set.

Reviewer: ```suggestion
You have to select at least one defining attribute to create a variant set. You can only select attributes as defining attributes that are neither multi-language nor multi-channel and assigned to the selected attribute set.
```

#### Prerequisites

- An attribute set has been created in DataHub, see [Create an attribute set](../../DataHub/Integration/02_ManageAttributeSets.md#create-an-attribute-set).
- The basic data of the variant set has been defined, see [Define the variant set basic data](#define-the-variant-set-basic-data).

#### Procedure

*Omni-Channel > Settings > Tab VARIANT SETS > Button Add*

![Formulas](../../Assets/Screenshots/Channels/Settings/VariantSets/Formulas.png "[Formulas]")

1. Click the ![Add](../../Assets/Icons/Plus01.png "[Add]") button in the *Defining attributes* box. This button is only displayed if an attribute set is selected.       
  The *Add defining attributes* view is displayed.

  ![Add defining attributes](../../Assets/Screenshots/Channels/Settings/VariantSets/AddDefiningAttributes.png "[Add defining attributes]")

> 2. Select the checkboxes of the attributes you want to be a defining attribute for the selected variant.   

Reviewer: oder for the selected variant set?

> 2. Select the checkboxes of the attributes you want to be a defining attribute for the selected variant.   

Reviewer: ```suggestion
2. Select the checkboxes of the attributes you want to be a defining attribute for the selected variant set.   
```

### Add changeable attributes

You can select changeable attributes for a variant set. If you want to define an attribute for the variants whose value may differ from its master product value, you have to define this attribute as a changeable attribute.

#### Prerequisites

- An attribute set has been created in DataHub, see [Create an attribute set](../../DataHub/Integration/02_ManageAttributeSets.md#create-an-attribute-set).
- The basic data of the variant set has been defined, see [Define the variant set basic data](#define-the-variant-set-basic-data).

#### Procedure

*Omni-Channel > Settings > Tab VARIANT SETS > Button Add*

![Formulas](../../Assets/Screenshots/Channels/Settings/VariantSets/Formulas.png "[Formulas]")

1. Click the ![Add](../../Assets/Icons/Plus01.png "[Add]") button in the *Changeable attributes* box. This button is only displayed if an attribute set is selected.         
  The *Add changeable attributes* view is displayed.

  ![Add changeable attributes](../../Assets/Screenshots/Channels/Settings/VariantSets/AddChangeableAttributes.png "[Add changeable attributes]")

> 2. Select the checkboxes of the attributes you want to be a changeable attribute for the selected variant.   

Reviewer: ```suggestion
2. Select the checkboxes of the attributes you want to be a changeable attribute for the selected variant set.   
```

Reviewer: Aus Interesse für mich: Warum nicht "for the selected variant set"? Die Attribute, die man hier auswählt, treffen alle Varianten im Variant set, oder?

[!!User interface Variant sets](../UserInterface/07b_VariantSets.md)
[!!Manage the variant sets in PIM](../../PIM/Integration/07_ManageVariantSets.md)
[!!Manage the variants in PIM](../../PIM/Operation/02_ManageVariants.md)
[!!Manage an attribute set in DataHub](../../DataHub/Integration/02_ManageAttributeSets.md)

# Manage a variant set

The variant sets in *Omni-Channel* are based on the same data model as in the Actindo *PIM* module. However, since the *Omni-Channel* and the Actindo *PIM* modules are independent of each other and *Omni-Channel* can also be used without the Actindo *PIM* module, the variant sets for products can also be managed within the *Omni-Channel* module.    
If both the *Omni-Channel* and the Actindo *PIM* modules are used, there is a predefined mapping that automatically applies the variant sets from *PIM* to *Omni-Channel*. In this case, you do not have to manage the variant sets in *Omni-Channel*, but in the Actindo *PIM* module, see [Manage the variant sets in PIM](../../PIM/Integration/07_ManageVariantSets.md).

> 

Reviewer: Vielleicht noch deutlicher/einfacher erklären, was ein variant set und ein variant ist (und Unterschied zwischen den beiden)?

5. For the next steps to create a variant set, follow the procedures below:   
  - [Add defining attributes](#add-defining-attributes)
  - [Add changeable attributes](#add-changeable-attributes)

  > [Info] At least one defining attribute is required to create a variant set. Changeable attributes can be selected optionally.

### Add defining attributes

You have to select at least one defining attribute to create a variant set. You can only select attributes as defining attributes that are neither multi-language nor multi-channel and that are assigned to the selected attribute set.

#### Prerequisites

- An attribute set has been created in DataHub, see [Create an attribute set](../../DataHub/Integration/02_ManageAttributeSets.md#create-an-attribute-set).
- The basic data of the variant set has been defined, see [Define the variant set basic data](#define-the-variant-set-basic-data).

#### Procedure

*Omni-Channel > Settings > Tab VARIANT SETS > Button Add*

![Formulas](../../Assets/Screenshots/Channels/Settings/VariantSets/Formulas.png "[Formulas]")

> 1. Click the ![Add](../../Assets/Icons/Plus01.png "[Add]") button in the *Defining attributes* box. This button is only displayed if an attribute set is selected.       

Reviewer: ```suggestion
1. Click the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button in the *Defining attributes* box. This button is only displayed if an attribute set is selected.       
```

    > [Info] You can edit the formula in the *Formula for SKU* field and include the defining attribute placeholder(s).

4. If desired, add changeable attributes to the variant set, see [Add changeable attributes](#add-changeable-attributes). Otherwise, click the [SAVE] button in the upper right corner.   
  The *Create variant set* view is closed. The new variant set has been saved and is displayed in the list of variant sets.

### Add changeable attributes

You can select changeable attributes for a variant set. If you want to define an attribute for the variants whose value may differ from its master product value, you have to define this attribute as a changeable attribute.

#### Prerequisites

- An attribute set has been created in DataHub, see [Create an attribute set](../../DataHub/Integration/02_ManageAttributeSets.md#create-an-attribute-set).
- The basic data of the variant set has been defined, see [Define the variant set basic data](#define-the-variant-set-basic-data).

#### Procedure

*Omni-Channel > Settings > Tab VARIANT SETS > Button Add*

![Formulas](../../Assets/Screenshots/Channels/Settings/VariantSets/Formulas.png "[Formulas]")

> 1. Click the ![Add](../../Assets/Icons/Plus01.png "[Add]") button in the *Changeable attributes* box. This button is only displayed if an attribute set is selected.         

Reviewer: ```suggestion
1. Click the ![Add](../../Assets/Icons/Plus01.png "[Add]") (Add) button in the *Changeable attributes* box. This button is only displayed if an attribute set is selected.         
```
